src/globals.c

```diff
@@ -205,11 +205,45 @@ static bool set_state_dir(char *path)
 	return true;
 }
 
+/* Sets the state_dir_cache global variable. If the path parameter is not
+ * NULL, state_dir_cache will be set to its value.
+ */
+bool set_state_dir_cache(char *path)
+{
+	if (!path) {
+		error("Statedir-cache shouldn't be set to NULL\n");
+		return false;
+	}
+
+	if (path[0] != '/') {
+		error("Statedir-cache must be a full path starting with '/', not '%c'\n", path[0]);
+		return false;
+	}
+
+	/* Prevent some disasters: since the statedir-cache can be destroyed and
+	 * reconstructed, make sure we never set those by accident and nuke the
+	 * system. */
+	if (!strcmp(path, "/") || !strcmp(path, "/var") || !strcmp(path, "/usr")) {
+		error("Refusing to use '%s' as a statedir-cache because it might be erased first\n", path);
+		return false;
+	}
+
+	free_string(&globals.state_dir_cache);
+	string_or_die(&globals.state_dir_cache, "%s", path);
+
+	return true;
+}
+
 static void set_default_state_dir()
 {
 	string_or_die(&globals.state_dir, "%s", STATE_DIR);
 }
 
+static void set_default_state_dir_cache()
```

Comment:
you missed adding a void as argument

Reply:
good catch. The PR is already merged, @castulo can you add that in a new commit?